sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs

```diff
@@ -158,28 +156,27 @@ public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder bui
                     {
                         var azureMonitorOptions = azureOptions.Get(Options.DefaultName);
 
-                        bool enableLogSampling = false;
-                        try
-                        {
-                            var enableLogSamplingEnvVar = Environment.GetEnvironmentVariable(EnableLogSamplingEnvVar);
-                            bool.TryParse(enableLogSamplingEnvVar, out enableLogSampling);
-                        }
-                        catch (Exception ex)
-                        {
-                            AzureMonitorAspNetCoreEventSource.Log.GetEnvironmentVariableFailed(EnableLogSamplingEnvVar, ex);
-                        }
-
-                        if (enableLogSampling)
+                        if (!azureMonitorOptions.DisableTraceBasedSamplingForLogs)
                         {
                             var azureMonitorExporterOptions = new AzureMonitorExporterOptions();
                             azureMonitorOptions.SetValueToExporterOptions(azureMonitorExporterOptions);
                             loggingOptions.AddProcessor(new LogFilteringProcessor(new AzureMonitorLogExporter(azureMonitorExporterOptions)));
                         }
                         else
                         {
-                            loggingOptions.AddAzureMonitorLogExporter(o => azureMonitorOptions.SetValueToExporterOptions(o));
+                            loggingOptions.AddAzureMonitorLogExporter(o =>
+                            {
+                                azureMonitorOptions.SetValueToExporterOptions(o);
+
+                                // Sampling is disabled by the user.
+                                // Set the SampleRatio of log exporter to 1.0.
+                                // The value will be used by the exporter to set SampleRate on the telemetry items.
+                                o.SamplingRatio = 1.0f;
+                            });
                         }
 
+                        // Sequence of adding processors does not matter here even if the filtering processor is used.
+                        // LogRecord will pass through each processor independent of other processors in the pipeline.
```

Comment:
nit: I think this comment is out of place. It describes LogRecord filtering processor, but it's listed before the EnableLiveMetrics. Should this instead be moved to line 163?